README.md

```diff
@@ -491,6 +492,62 @@ r.Use(amw.Middleware)
 
 Note: The handler chain will be stopped if your middleware doesn't call `next.ServeHTTP()` with the corresponding parameters. This can be used to abort a request if the middleware writer wants to. Middlewares _should_ write to `ResponseWriter` if they _are_ going to terminate the request, and they _should not_ write to `ResponseWriter` if they _are not_ going to terminate it.
 
+### CORSMethodMiddleware
+
+`CORSMethodMiddleware` intends to make it easier to strictly set the `Access-Control-Allow-Methods` response header. You will still need to use your own CORS handler to set the other CORS headers such as `Access-Control-Allow-Origin`.
+
+`CORSMethodMiddleware` will set the `Access-Control-Allow-Methods` header to all the method matchers (e.g. `r.Methods(http.MethodGet, http.MethodPut, http.MethodOptions)` -> `Access-Control-Allow-Methods: GET,PUT,OPTIONS`) on a route. Note that there must be an `OPTIONS` method matcher for the middleware to set the headers.
```

Comment:
I would break this out into bullets to make it more readable.

* You will still need to use your own CORS handler to set the other CORS headers such as `Access-Control-Allow-Origin` - e.g. such as [rs/cors](https://github.com/rs/cors)
* The middleware will set the `Access-Control-Allow-Methods` header to all the method matchers (e.g. `r.Methods(http.MethodGet, http.MethodPut, http.MethodOptions)` -> `Access-Control-Allow-Methods: GET,PUT,OPTIONS`) on a route. 
* If you do not specify any methods, then <behaviour>

> **Important**: there must be an `OPTIONS` method matcher - `.Methods(http.MethodOptions)` - for the middleware to set the CORS response headers.

Reply:
Good idea, will do.